cli/src/trusted_operation.rs

```diff
@@ -188,7 +196,10 @@ fn send_indirect_request(
 				};
 
 				if confirmed_block_hash == block_hash {
-					return Ok(Some(block_hash.encode()))
+					// encode and decode to target type, this should probably read value from parachain event and
+					// return that result instead of block hash
+					let value = decode_response_value(&mut block_hash.encode().as_slice())?;
```

Comment:
Yes, in theory. For this, extrinsic, however, it is not really IMO as the extrinsics can't fail due to the onchain logic, but good thought!

Reply:
Removed that comment.